src/Sarif/CodeGenHints.json

```diff
@@ -434,6 +442,30 @@
       }
     }
   ],
+  "threadFlowLocation": [
+    {
+      "kind": "BaseTypeHint",
+      "arguments": {
+        "baseTypeNames": [
+          "PropertyBagHolder"
+        ]
+      }
+    }
+  ],
+  "ThreadFlowLocation.State": [
+    {
+      "kind": "DictionaryHint"
+    }
+  ],
+  "ThreadFlowLocation.Importance": [
+    {
+      "kind": "EnumHint",
+      "arguments": {
+        "typeName": "ThreadFlowLocationImportance",
+        "description": "Values specifying the importance of an \"threadFlowLocation\" within the \"codeFlow\" in which it occurs"
+      }
```

Comment:
>} [](start = 6, length = 1)

Bug: You need to supply `arguments.memberNames`.

Reply:
Note that I only moved this code and did not author it. (for ThreadFlowLocation.Importance). The enum is created in my clean build, with correct enum values. Are you sure that your autogen doesn't provide correct behavior by default? perhaps arguments.memberNames is only required if we insist on creating an additional member name that's not explicit in the schema. Or if some renaming heuristics are broken?

---
In reply to: [208046177](https://github.com/Microsoft/sarif-sdk/pull/989#discussion_r208046177) [](ancestors = 208046177)